spec/tasks/publishing_api_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe 'rake publishing_api:*', type: task do
+  let(:consumer) { double('message_consumer', run: nil) }
+
+  it 'start listener for PublishingAPI events' do
```

Comment:
I would prefer to test the processor more extensively than test these rake tasks. The rake tasks' only responsibility is to make a single function call, and this code will rarely change, whereas the processor itself will change a lot.

I think if the consumer is not running in production, the most likely scenario is it crashed because of an unhandled exception.

Reply:
I find value in unit testing the rake task, because as you say, it main responsibility is to make a function call. I like it because we are being proactive towards preventing the Jenkins job from failing and having to be reactive in fixing it.

When it comes to the testing, agree that we need an integration tests for the processor, but not sure if this should live here, as this is just an entry point for the process. I would be really happy to create another PR to add an integration test that cover the basic scenarios.